Release CI workflows failing due to character set issues

### Description

The `release-automation.yml`, `release-integration.yml`, and the `release-production.yml` workflows are missing a necessary `LC_ALL=C.UTF-8` environment variable declaration to support rosetta testing of filenames containing specific UTF-8 escapes. 

### Steps to reproduce

1. Trigger any of the aforementioned workflows. 

### Additional context

```shell
> Task :importer:compileJava

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':rosetta:setup'.
> Malformed input or input contains unmappable characters: /runner/_work/hedera-mirror-node/hedera-mirror-node/.gradle/go/test/fixedbugs/issue27836.dir/?foo.go

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org./

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.


BUILD FAILED in 1m 2s
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.6/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
50 actionable tasks: 28 executed, 22 from cache
WARNING: The following functionality has been deprecated and will be removed in the next major release of the Develocity Gradle plugin:
- The deprecated "gradle.enterprise.testretry.enabled" system property has been replaced by "develocity.testretry.enabled"

Publishing build scan...
https://gradle.com/s/fbwj7vqs4mwiu
```

### Hedera network

other

### Version

latest

### Operating system

Linux

--- a/.github/workflows/release-automation.yml
+++ b/.github/workflows/release-automation.yml
@@ -14,6 +14,9 @@ defaults:
   run:
     shell: bash
 
+env:
+  LC_ALL: C.UTF-8
+
 jobs:
   release:
     name: Release

--- a/.github/workflows/release-integration.yml
+++ b/.github/workflows/release-integration.yml
@@ -2,14 +2,18 @@ name: Release Integration
 
 on:
   push:
-    branches: [main]
+    branches:
+      - 'main'
 
 permissions:
-  contents: read
+  contents: write
 
 defaults:
   run:
     shell: bash
+
+env:
+  LC_ALL: C.UTF-8
 
 jobs:
   publish:

--- a/.github/workflows/release-production.yml
+++ b/.github/workflows/release-production.yml
@@ -2,7 +2,8 @@ name: Release Production
 
 on:
   push:
-    tags: [v*]
+    tags:
+      - 'v*'
 
 permissions:
   contents: read
@@ -10,6 +11,9 @@ permissions:
 defaults:
   run:
     shell: bash
+
+env:
+  LC_ALL: C.UTF-8
 
 jobs:
   image:
@@ -95,13 +99,15 @@ jobs:
     name: Publish charts
     needs: image
     runs-on: [self-hosted, linux, large, ephemeral]
+    permissions:
+      contents: write
     steps:
       - name: Harden Runner
         uses: step-security/harden-runner@63c24ba6bd7ba022e95695ff85de572c04a18142 # v2.7.0
         with:
           egress-policy: audit
 
-      - name: Checkout
+      - name: Checkout Code
         uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
 
       - name: Publish helm chart

